application/src/objects/abstract-marker.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var Phaser = require('phaser');
+var _      = require('underscore');
+
+var AbstractMarker = function(state, x, y, key, angle, length)
+{
+    var validAngles;
+
+    Phaser.Sprite.apply(this, [state.game, x, y, key]);
+
+    this.state = state;
+
+    // This will make the added width add to both sides of the sprite
+    // equally. Without it, the x and y passed in will specify the position
+    // of the very left edge of the marker
+    this.anchor.setTo(0.5, 0.5);
+
+    this.height = 32;
+    this.width  = length;
+
+    validAngles = [0, 90, 180, 270];
+    if (! _(validAngles).contains(angle)) {
+        throw new Error('Invalid marker angle. Must be one of: ' + validAngles.join(', '));
+    }
```

Comment:
I think there is a way to do this within the physics system. Apparently there's a `shape.sensor` attribute that lets you make a shape that doesn't create impacts, but can detect whether something else is touching/overlapping it. But for now we can just treat this as a constraint to our track design, and if it annoys either of us enough to fix it we can address it then.